test/renate/memory/test_storage.py

```diff
@@ -21,25 +22,72 @@ def nested_tensors_equal(t1, t2):
         return all(nested_tensors_equal(t1[key], t2[key]) for key in t1.keys())
 
 
-@pytest.mark.parametrize("length", [0, 1, 10])
-@pytest.mark.parametrize(
-    "data_point",
-    [
-        torch.tensor(1),
-        torch.ones(3),
-        (torch.ones(3), torch.tensor(1)),
-        ({"a": torch.ones(2, 3, 3), "b": torch.zeros(2)}, torch.tensor(4)),
-        {"a": (torch.ones(2, 3, 3), torch.zeros(2)), "b": torch.tensor(2)},
-    ],
-)
-def test_storage(tmpdir, length, data_point):
-    """Tests the memory-mapped tensor storage for different nested tensor structures."""
-    storage = MemoryMappedTensorStorage(tmpdir, data_point, length)
-    for i in range(length):
-        storage[i] = data_point
+def make_dataset_same_sizes(dataset_length, return_type):
+    class CustomDataset(Dataset):
+        def __init__(self, x, y, return_type="tensor") -> None:
+            self.x, self.y = x, y
+            self.return_type = return_type
+
+        def __getitem__(self, index):
+            if self.return_type == "tensor":
+                return self.x[index]
+            elif self.return_type == "tuple":
+                return (self.x[index], self.y[index])
+            elif self.return_type == "dict":
+                return {"x": self.x[index], "y": self.y[index]}
+
+        def __len__(self):
+            return self.x.shape[0]
+
+    data_x = torch.rand(dataset_length, 3, 32, 32)
+    data_y = torch.randint_like(data_x, 5)
+
+    return CustomDataset(data_x, data_y, return_type=return_type)
+
+
+@pytest.mark.parametrize("length", [1, 10])
+@pytest.mark.parametrize("return_type", ["tuple", "dict", "tensor"])
+@pytest.mark.parametrize("StorageClass", [FileTensorStorage])
+def test_memory_storage_same_size(tmpdir, length, return_type, StorageClass):
+    ds = make_dataset_same_sizes(length, return_type)
+    storage = StorageClass(tmpdir)
+    storage.dump_dataset(ds)
 
     del storage
-    storage = MemoryMappedTensorStorage(tmpdir, data_point, length)
+    storage = StorageClass(tmpdir)
+    for i in range(length):
+        assert nested_tensors_equal(storage[i], ds[i])
```

Comment:
test logic is identical to test logic for "different size" besides the very first line. reuse validation logic